pkg/reconciler/metric/metric.go

```diff
@@ -60,8 +60,13 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		return errors.Wrapf(err, "failed to fetch metric %q", key)
 	}
 
+	metric.Status.InitializeConditions()
+
 	if err := r.collector.CreateOrUpdate(metric); err != nil {
+		metric.Status.MarkMetricNotReady("CreateOrUpdateFailed", "Metric creation or update has failed.")
```

Comment:
Can we inspect the error to determine whether it's failed or not ready?
Also it's Collector that failed, not the metric itself.